wm_dll.c

```diff
@@ -13,20 +18,61 @@ __declspec(dllexport) LRESULT CALLBACK ShellProc(int code, WPARAM wparam, LPARAM
 	return CallNextHookEx(NULL, code, wparam, lparam);
 }
 
+__declspec(dllexport) LRESULT HotkeyProc(int code, WPARAM wparam, LPARAM lparam) {
```

Comment:
Why have you decided to place this in the DLL? this can be in the main executable.

Reply:
I was under the impression that all of the final window handling will be done in the DLL, it will have to be decided what the DLL is for, because just as easily could move the ShellProc hook to the lightwm.exe and get rid of the DLL entirely.

Reply:
Since the DLL is added into many processes on the system I want to keep the logic in it as minimal as possible - that is why it is only signaling an event and the actual handling occurs in the executable.
The ShellProc can't move to the main executable and has to be in a DLL since the SetWindowsHookEx function requires it to come from a DLL if you want to place this hook system wide (see parameter lpfn and dwThreadId in https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-setwindowshookexw ).

Reply:
This requirement about the hook function is basically the reason I made this DLL

Reply:
If you notice in the table WH_SHELL is Thread or Global meaning it can be installed both in the installing thread or a global thread. I just made a testing branch where I removed the DLL entirely and it works just fine. 

https://github.com/DemiRom/lightwm/tree/testing/dll-removal

Reply:
Try created or destroying a window after the change, I tried as well now installing the hook only on the current thread, and it doesn't catch any additional windows

Reply:
Okay I see, sometimes I run it without the DLL it does work but most times it doesn't, I guess that all makes sense now. So its settled the DLL is only for the WH_SHELL hook until another global hook is required.